making_change/making_change.py

```diff
@@ -3,8 +3,14 @@
 import sys
 
 def making_change(amount, denominations):
-  pass 
-
+  if amount == 0:
+    return 1
+  if amount < 0:
+    return 0
+  if amount > 0 and len(denominations) == 0:
+    return 0
+  else:
+    return making_change(amount - denominations[-1], denominations) + making_change(amount, denominations[:-1])
```

Comment:
very efficient clean method, no need for external data structures and you avoid repetition errors by iterating through the sorted denominations list in reverse order.